runtime/vm/ContinuationHelpers.cpp

```diff
@@ -923,14 +923,110 @@ preparePinnedVirtualThreadForUnmount(J9VMThread *currentThread, j9object_t syncO
 	UDATA result = J9_OBJECT_MONITOR_YIELD_VIRTUAL;
 	J9ObjectMonitor *syncObjectMonitor = NULL;
 	J9ObjectMonitor *enteredMonitorsList = NULL;
-	j9objectmonitor_t lock = 0;
-	j9object_t continuationObj = NULL;
 	UDATA monitorCount = 0;
 	J9JavaVM *vm = currentThread->javaVM;
 
 	if (NULL != syncObj) {
+		j9objectmonitor_t volatile *lwEA = VM_ObjectMonitor::inlineGetLockAddress(currentThread, syncObj);;
+		j9objectmonitor_t lock = J9_LOAD_LOCKWORD(currentThread, lwEA);
+		omrthread_monitor_t monitor = NULL;
+		
 		enterVThreadTransitionCritical(currentThread, (jobject)&currentThread->threadObject);
+
+		if (J9_LOCK_IS_INFLATED(lock)) {
+			syncObjectMonitor = J9_INFLLOCK_OBJECT_MONITOR(lock);
+		} else {
+			if (isObjectWait) {
+				syncObjectMonitor = objectMonitorInflate(currentThread, syncObj, lock);
+				if (NULL == syncObjectMonitor) {
+					result = J9_OBJECT_MONITOR_OOM;
+					goto done;
+				}
+			} else {
+				/* This must be a monitor enter case, so this implies that a monitor entry was
+				 * created as the non-blocking path would have failed.
+				 * Since a monitor can only be inflated by a thread that owns it, we can't direct
+				 * inflate the blocking monitor.
+				 */
+restart:
+#if defined(J9VM_THR_LOCK_RESERVATION)
+				{
+					while (OBJECT_HEADER_LOCK_RESERVED == (lock & (OBJECT_HEADER_LOCK_RESERVED | OBJECT_HEADER_LOCK_INFLATED))) {
+						cancelLockReservation(currentThread);
+						/* calculate the new lock word, since the object may have moved. */
+						syncObj = J9VMTHREAD_BLOCKINGENTEROBJECT(currentThread, currentThread);
+						lwEA = VM_ObjectMonitor::inlineGetLockAddress(currentThread, syncObj);
+						lock = J9_LOAD_LOCKWORD(currentThread, lwEA);
+						if (VM_ObjectMonitor::inlineFastInitAndEnterMonitor(currentThread, lwEA)) {
+							result = (UDATA)syncObj;
+							goto done;
+						}
+					}
+				}
+#endif /* J9VM_THR_LOCK_RESERVATION */
+				syncObjectMonitor = monitorTableAt(currentThread, syncObj);
+				monitor = syncObjectMonitor->monitor;
+
+				/* Try acquire the inflated monitor */
+				if (0 == omrthread_monitor_try_enter(monitor)) {
+					/* if the INFLATED bit is set, then it is already inflated and we own the object monitor. */
+					if (J9_ARE_ANY_BITS_SET(((J9ThreadMonitor*)monitor)->flags, J9THREAD_MONITOR_INFLATED)) {
+						currentThread->ownedMonitorCount += 1;
+						result = (UDATA)syncObj;
+						goto done;
+					}
+
+					/* Loop until either lock is inflated or FLC bit is set. */
+					while (!J9_LOCK_IS_INFLATED(lock)) {
+						/* The monitor isn't inflated yet, but we can update the lockword now. */
+						j9objectmonitor_t newLockword = 0;
+						UDATA count = J9_FLATLOCK_COUNT(lock);
+						if (0 == count) {
+							/* lock is unlocked, so try to directly acquire it as a flatlock. */
```

Comment:
Please start each sentence with a capital; see also lines 988, 1000, 1008 and 1016.